### PR TITLE
Add UUID api and improve join logic

### PR DESCRIPTION
**Pull request by Meierschlumpf**
---
name: Add UUID api and improve join logic
about: In this pullrequest the uuid api is added to allow the detection of name changes. Additionally the join logic was improved.
assignees: @Meierschlumpf 

---

The new added uuid api allows to save all uuids of the joined players and matching them with their player id. By that it is possible to find out which ID a player had before he changed his name. 

What needs to be done for full functionallity:
- Run the function `api:uuid/init` to initialize the data storage.
- Run for each player that has already been registered the function `api:uuid/add`.

### CODE REVIEW

File: data/api/functions/uuid/check/index.mcfunction [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#
```
Comment (Meierschlumpf) [inline data/api/functions/uuid/check/index.mcfunction]: Remove old comment

File: data/vm/functions/5t.mcfunction [hunk 2]
```diff
@@ -1,13 +1,4 @@
...
-scoreboard players set @a[team=] rank 90
-scoreboard players set @a[team=] l 1
-team join 90player @a[team=]
-
-scoreboard players set @a[scores={rejoin=1..}] l 1
-execute as @a[scores={rejoin=1..}] run tellraw @s [{"text":"Welcome back, ","color":"green"},{"selector":"@s"},{"text":"!"}]
-scoreboard players reset @a[scores={rejoin=1..}] rejoin
-
-execute at @a[tag=lobby,nbt={SelectedItem:{id:"minecraft:compass"}}] unless entity @e[type=chest_minecart,tag=lobby_gui,distance=..5] run function vm:gui/create
+execute at @a[tag=lobby,nbt={SelectedItem: {id: "minecraft:compass"}}] unless entity @e[type=chest_minecart,tag=lobby_gui,distance=..5] run function vm:gui/create
```
Comment (Meierschlumpf) [inline data/vm/functions/5t.mcfunction]: Reset spacing changes

File: data/vm/functions/tick.mcfunction [hunk 3]
```diff
@@ -1,20 +1,21 @@
-execute as @a[tag=!registered] run function api:players/create/current
+execute as @a[tag=!registered] run function vm:join_server/first
+execute as @a[scores={rejoin=1..}] run function vm:join_server/index
 
 execute as @a[tag=skywars,nbt={OnGround: 0b}] store result score @s apply_damage run data get entity @s FallDistance
-execute as @a[scores={apply_damage=1..},gamemode=!creative,gamemode=!spectator,nbt={OnGround:1b}] run scoreboard players remove @s apply_damage 3
-execute as @a[scores={apply_damage=1..},gamemode=!creative,gamemode=!spectator,nbt={OnGround:1b}] run function vm:damage
+execute as @a[scores={apply_damage=1..},gamemode=!creative,gamemode=!spectator,nbt={OnGround: 1b}] run scoreboard players remove @s apply_damage 3
```
Comment (Meierschlumpf) [inline data/vm/functions/tick.mcfunction]: Reset spacing changes

File: data/api/functions/uuid/check/without_id.mcfunction [hunk 4]
```diff
@@ -0,0 +1,7 @@
+scoreboard players set #foundId temp 0
+execute store result score #lenght temp run data get storage api:uuid list
+data modify storage api:uuid temp set from storage api:uuid list
+execute unless score #lenght temp matches 0 run function api:uuid/check/without_id/repeat
```
Comment (DEVTomatoCake) [inline data/api/functions/uuid/check/without_id.mcfunction]:
```suggestion
execute store result score #length temp run data get storage api:uuid list
data modify storage api:uuid temp set from storage api:uuid list
execute unless score #length temp matches 0 run function api:uuid/check/without_id/repeat
```

File: data/api/functions/uuid/check/without_id/repeat.mcfunction [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+function api:uuid/check/compare_uuids
+
+execute if score #matches temp matches 1 store result score #foundId temp run data get storage api:uuid temp[0].id
+
+scoreboard players remove #lenght temp 1
+
+data remove storage api:uuid temp[0]
+
+execute unless score #lenght temp matches ..0 if score #matches temp matches 0 run function api:uuid/check/without_id/repeat
```
Comment (DEVTomatoCake) [inline data/api/functions/uuid/check/without_id/repeat.mcfunction]:
```suggestion
scoreboard players remove #length temp 1

data remove storage api:uuid temp[0]

execute unless score #length temp matches ..0 if score #matches temp matches 0 run function api:uuid/check/without_id/repeat
```

**Review by DEVTomatoCake — CHANGES_REQUESTED** [review]
Small typo, looks good otherwise

**Review by DEVTomatoCake — APPROVED** [review]

**Pull request merged**